File: app/src/main/java/com/hackerx/todonotesapp/Adapter/NotesAdapter.java
package com.hackerx.todonotesapp.Adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.textview.MaterialTextView;
import com.hackerx.todonotesapp.Notes;
import com.hackerx.todonotesapp.R;

import java.util.List;

import ClickListener.ItemClickListener;

public class NotesAdapter extends RecyclerView.Adapter<NotesAdapter.ViewHolder> {

    private List<Notes> listNotes;
    private ItemClickListener itemClickListener;
    public NotesAdapter(List<Notes> list, ItemClickListener itemClickListener){
        this.listNotes = list;
        this.itemClickListener = itemClickListener;
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        MaterialTextView title , description;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            title = itemView.findViewById(R.id.card_title_text);
            description = itemView.findViewById(R.id.card_desc_text);
        }
    }


    @NonNull
    @Override
    public NotesAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.notes_adapter_layout,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull NotesAdapter.ViewHolder holder, int position) {
        String title,description;
       final  Notes notes = listNotes.get(position);
        title = notes.getTitle();
        description = notes.getDescription();

        holder.title.setText(title);
        holder.description.setText(description);

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                itemClickListener.onClick(notes);
            }
        });
    }


    @Override
    public int getItemCount() {
        return listNotes.size();
    }


}

File: app/src/main/java/com/hackerx/todonotesapp/Adapter/NotesAdapter.kt
package com.hackerx.todonotesapp.Adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CompoundButton
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.checkbox.MaterialCheckBox
import com.google.android.material.textview.MaterialTextView
import com.hackerx.todonotesapp.ClickListener.ItemClickListener
import com.hackerx.todonotesapp.db.Notes
import com.hackerx.todonotesapp.R

class NotesAdapter(val list: List<Notes>, val itemClickListener:ItemClickListener) : RecyclerView.Adapter<NotesAdapter.ViewHolder>() {

    inner class ViewHolder(itemView:View) : RecyclerView.ViewHolder(itemView)
    {
        val texttitle:MaterialTextView = itemView.findViewById(R.id.card_title_text)
        val textdescription:MaterialTextView = itemView.findViewById(R.id.card_desc_text)
        val checkBoxMarkStatus:MaterialCheckBox = itemView.findViewById(R.id.card_checkbox)
    }
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NotesAdapter.ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.notes_adapter_layout,parent,false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int {
        return list.size
    }

    override fun onBindViewHolder(holder: NotesAdapter.ViewHolder, position: Int) {
        val notes  = list[position]
        val title = notes.title
        val description = notes.description
        //bind data to the recyclerview adapter
        holder.texttitle.text = title
        holder.textdescription.text = description
        holder.checkBoxMarkStatus.isChecked = notes.isTaskCompleted
        holder.itemView.setOnClickListener(object : View.OnClickListener{
            override fun onClick(p0: View?) {
                itemClickListener.onClick(notes)
            }

        })
        holder.checkBoxMarkStatus.setOnCheckedChangeListener(object : CompoundButton.OnCheckedChangeListener{
            override fun onCheckedChanged(buttonView: CompoundButton?, isChecked: Boolean) {
                notes.isTaskCompleted = isChecked
                itemClickListener.onUpdate(notes)
            }

        })

    }

}